### ISSUE TITLE
alpha_factor in focal loss

### ISSUE THREAD

**Issue by MiladShahidi** [tensorflow/addons#1863]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS 10.14.6
- TensorFlow version and how it was installed (source or binary): 2.1.0
- TensorFlow-Addons version and how it was installed (source or binary): binary 0.10.0
- Python version: 3.6
- Is GPU used? (yes/no): no

The focal loss should be equal to cross-entropy when gamma=0 and alpha=1. However, this is not the case in the current code due to [this line of code](https://github.com/tensorflow/addons/blob/f3d0416fbe3f4c607f4c1ebc2b7554a941ff02a2/tensorflow_addons/losses/focal_loss.py#L129). I'm not sure why y_true is in there. When you replace alpha_factor with alpha in the return statement, (and set gamma=0 and alpha=1) the loss becomes identical to cross-entropy, as expected.

**Comment by MiladShahidi** [comment]
Sorry, misread the paper. Feel free to remove this if possible.

**Comment by FirefoxMetzger** [comment]
Funny, I was just about to open an issue about the alpha value as well, but on a different note.

From my understanding of the work, alpha is used to weight samples (in this case by a class dependent constant). Keras losses already allow that via `sample_weight`. Considering this, I think the implementation should not use `alpha` and leave this to `sample_weight`/`class_weight`. In fact, -as the loss inherits from `tf.keras.losses.Loss` it already does this effectively implementing alpha twice.
